src/rosdiscover/observer/ros1.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+__all__ = ('ROS1Observer',)
+
+import typing
+from typing import Collection, Dict, Set
+
+from loguru import logger
+from roswire import AppInstance, ROS1
+from roswire.common import SystemState
+
+from .nodeinfo import NodeInfo
+from .observer import Observer
+from ..interpreter import NodeContext, ParameterServer, SystemSummary
+
+if typing.TYPE_CHECKING:
+    from ..config import Config
+
+_NODES_TO_FILTER_OUT = ('/rosout',)
+_TOPICS_TO_FILTER_OUT = ('/rosout', '/rosout_agg')
+_SERVICES_TO_FILTER_OUT = ('set_logger_level', 'get_loggers')
+
+
+class ROS1Observer(Observer):
+
+    def __init__(self, app: AppInstance, config: 'Config') -> None:
+        super().__init__(app, config)
+
+    def observe(self) -> SystemSummary:
+        """Observe the state of the running system and produce a summery of the archtiecture."""
+        nodecontexts: Set[NodeContext] = set()
+        with self._app_instance.ros1() as ros:
+            info = ros.state
+            nodes = self._transform_info(info)
+            for node in nodes:
+                nodecontext = self._make_node_context(node, ros)
+                nodecontexts.add(nodecontext)
+
+        return self._summarise(nodecontexts)
+
+    def _summarise(self, contexts: Collection[NodeContext]) -> SystemSummary:
+        """Produces an immutable description of the system architecture.
+
+        Parameters
+        ----------
+        contexts: Collection[NodeContext]
+            A collection of contexts that are to be summarised.
+
+        Returns
+        -------
+        SystemSummary
+            A summary of the system architecture.
+        """
+        summaries = [node.summarise() for node in contexts]
+        node_to_summary = {s.fullname: s for s in summaries}
+        return SystemSummary(node_to_summary)
+
+    def _make_node_context(self, node: NodeInfo, ros: ROS1) -> NodeContext:
+        """
+        Construct a NodeContext used in rosdiscover from the processed information from roswire.
+
+        Parameters
+        ----------
+        node: NodeInfo
+            The processed node information from roswire
+        ros: ROS1
+            Information about the state (used for getting types)
+
+        Returns
+        -------
+            A NodeContext
```

Comment:
It's fine to omit parts of the docstring for non-public methods if those parts are obvious.